src/v/cluster/tm_stm.cc

```diff
@@ -327,18 +327,48 @@ tm_stm::do_update_tx(tm_transaction tx, model::term_id term) {
 
     auto r = co_await replicate_quorum_ack(term, std::move(batch));
     if (!r) {
+        vlog(
+          txlog.info,
+          "got error {} on updating tx:{} pid:{} etag:{} tx_seq:{}",
+          r.error(),
+          tx.id,
+          tx.pid,
+          tx.etag,
+          tx.tx_seq);
+        if (_c->is_leader() && _c->term() == term) {
+            co_await _c->step_down(
+              "txn coordinator update_tx replication error");
+        }
         if (r.error() == raft::errc::shutting_down) {
             co_return tm_stm::op_status::timeout;
         }
         co_return tm_stm::op_status::unknown;
     }
 
-    if (!co_await wait_no_throw(
-          model::offset(r.value().last_offset()), _sync_timeout)) {
+    auto offset = model::offset(r.value().last_offset());
+    if (!co_await wait_no_throw(offset, _sync_timeout)) {
```

Comment:
same comment as before, whats the intent with stepping down

Reply:
to guarantee that the state tm_stm is operating is up to date, we may do this without stepping down but imho it's easier to start with simpler system and gradually make it more complex